src/sentry/static/sentry/app/views/projectFilters.jsx

```diff
@@ -198,38 +200,212 @@ const LegacyBrowserFilterRow = React.createClass({
   }
 });
 
+const ProjectFiltersSettingsForm = React.createClass({
+  propTypes: {
+    orgId: React.PropTypes.string.isRequired,
+    projectId: React.PropTypes.string.isRequired,
+    initialData: React.PropTypes.object.isRequired
+  },
+
+  mixins: [ApiMixin],
+
+  getInitialState() {
+    let formData = {};
+    for (let key of Object.keys(this.props.initialData)) {
+      if (key.lastIndexOf('filters:') === 0) {
+        formData[key] = this.props.initialData[key];
+      }
+    }
+    return {
+      hasChanged: false,
+      formData: formData,
+      errors: {},
+    };
+  },
+
+  onFieldChange(name, value) {
+    let formData = this.state.formData;
+    formData[name] = value;
+    this.setState({
+      formData: formData,
```

Comment:
Since the `formData` reference doesn't change here, this is not guaranteed to re-render.

Do:

```javascript
this.setState({
  ...formData,
  hasChanged: true
});
```